Diagnostics/Get-VSwitchPathCost.ps1

```diff
@@ -1,293 +1,254 @@
 #
-function GetAverageFromGetCounterJob($JobId)
+function Get-CounterAverage([String[]] $statArray)
 {
-    Stop-Job $JobId
-    $statArray = Receive-Job $JobId
-    Remove-Job $JobId
-
-    $avg = 0
-    $count=0
-    $endIndex = $statArray.Count - 1 - $CoolDownDurationInSeconds
-    $startIndex = $endIndex - $DurationInSeconds
-            
-    $message = ""
-    for ($i=$startIndex; $i -le $endIndex; $i++)
-    {
-        $value = $statArray[$i].Readings.Split(":")[-1]
-        $value = $value.Trim()
+    $sum = 0
+    $count = 0
+    $endIndex = $statArray.Count - 1 - $Cooldown
+    $startIndex = $endIndex - $Duration
 
-        $value = [int64]$value
-        $avg += $value
+    for ($i = $startIndex; $i -le $endIndex; $i++)
+    {
+        $value = $statArray[$i].Split(":")[1]
+        $value = [Int64] $value.Trim()
 
-        $message = $message + [string]$value + ", "
+        $sum += $value
         $count++
     }
 
-    $avg = $avg / $count
-
-    return $avg
+    return $sum / $count
 }
 
 
-###############################################################
-### Start program #############################################################
-###############################################################
-
+#
+# Start program
+#
 
-#######################################
+#
 # Sanity check the parameters & host setup
-#######################################
+#
 
-$switch = Get-Vmswitch $SwitchName
-if ($switch -eq $null)
+if (IsHyperThreadingEnabled)
 {
-   Log "ERROR: Switch $SwitchName does not exist"
+    Log "WARNING: Hyper-Threading is enabled. Hyper-Threading should be disabled for accurate path cost calculation."
+    if (-not $Force)
+    {
+        $options = @(
+            (New-Object Management.Automation.Host.ChoiceDescription "&Yes","Continue"),
+            (New-Object Management.Automation.Host.ChoiceDescription "&No","Exit")
+        )
+
+        $result = $Host.UI.PromptForChoice("", "Do you want to continue?", $options, 1)
+        if ($result -eq 1)
+        {
+            exit 1
+        }
+    }
 }
 
-if ($switch.Count -gt 1)
+if ($BaseCpuNumber -gt $(Get-MaxCpuNumber))
 {
-    Log "ERROR: There are more than one vswitch with the name $SwitchName"
+    Log "ERROR: BaseCpuNumber is greater than Max CPU Number."
 }
 
-if ($MaxCpuNumber -eq $null)
+if (-not $PSBoundParameters.ContainsKey("MaxCpuNumber"))
 {
-    $cpuPerNuma = GetNumCpusPerNUMA
-    $numaCount = GetNumaCount
-    $MaxCpuNumber = $cpuPerNuma * $numaCount - 1
+    $MaxCpuNumber = Get-MaxCpuNumber
     Log "Set MaxCpuNumber to $MaxCpuNumber"
 }
-
-if ($MaxCpuNumber -lt $BaseCpuNumber)
-{
-   Log "ERROR: Max CPU number is less than base CPU number"
-}
-
-#
-# Verify HyperThreading is disabled
-#
-if (IsHyperThreadingEnabled)
+elseif (IsHyperThreadingEnabled)
 {
-    Log "ERROR: HyperThreading is enabled. Please disable HyperThreading before starting the test."
+    # force MaxCpuNumber to odd value with HT so the entire core is measured
```

Comment:
maybe print a warning?

Reply:
Just realized that BaseCpuNumber should be forced to even value as well.